### DISCUSSION TITLE
[Desktop/Web] Go back to previously selected note when cancelling search

### DISCUSSION THREAD

**Discussion by christianhans** [standardnotes/forum#2597] [Suggestions]
Sometimes, when writing a specific note I want to look up or copy information stored in another note.

1. Since I don't know where exactly the note is located I search for it using the SN search box
2. Search finds and opens the other note, and I copy text for example
3. I want to go back to my previous note and hit the X icon next to the search box
4. SN stays on the note that I searched for and where I copied the text (not the previously edited note)
5. I have to manually go back to my previously edited note (which is the very top one in case I didn't modify the searched note)

When cancelling search, I think it would be a lot better user experience if StandardNotes restored the "state" and opened the note that has been selected before search was triggered. This would also entail placing the cursor where it has been before. Apple Notes opens the previously active note when cancelling search (but doesn't restore cursor position).

**Comment by atmoio** [comment]
Hmm..I can see how this can be useful. But also I wonder if some people might prefer to keep the searched for note as active when cancelling. Can you document how other popular note-taking apps handle this? I tested Apple Notes using their web app at icloud.com, and they handle it perhaps even worse: when you press X, it goes back to the first note in the list.

**Comment by christianhans** [comment]
Here is a video of how Apple Notes behaves:

https://user.fm/files/v2-40064db77285e858c063708e27592a6b/Screen%20Recording%202019-04-28%20at%202.12.49%20PM.mov

**Comment by atmoio** [comment]
Well, you only had 1 note in the list there ;) So it's hard to gauge its behavior. Trying having 4-5 notes in the list, select the 3rd one, search, cancel, and it will most likely go back and select the first note in the list rather than the 3rd.

**Comment by christianhans** [comment]
I think it will select the 3rd one even in that case. See video:

https://www.dropbox.com/s/uqxx118ruiw69t8/Screen%20Recording%202019-05-17%20at%2020.28.48.mov?dl=0
